domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs

```diff
@@ -42,12 +42,10 @@ public ContractServiceImpl(IMeteringPointsClient client, ICertificateIssuingCont
 
         var documents = await repository.GetByGsrn(gsrn, cancellationToken);
 
-        foreach (var x in documents)
+        var documentExist = CheckPeriod(documents, startDate, endDate);
```

Comment:
It seems like check period does a bit more than actually just checking a period.
```Suggestion
        foreach (var document in documents)
        {
            if ((isContractAlreadyExisting(document))
            {
                return new ContractAlreadyExists(document);
            }
        }

private bool isContractAlreadyExisting(document, startDate, endDate) 
{
    return startDate > document.StartDate && startDate < document.EndDate) || (endDate > document.StartDate && endDate < document.EndDate
}

```
Then we also avoid the null check